scripts/ontology_to_ontology_mapping_tool/ontology_to_ontology_mapping_tool.py

```diff
@@ -0,0 +1,171 @@
+        for code in ACCEPTED_ONTOLOGIES:
+            print(code)
+        print('\n')
+        sys.exit(1)
+    else:
+        print("The input file has the following ontology code columns: ")
+        for code in matched_headers:
+            print(code)
+        source_code = input("Please specify the source code column to map by: ")
+        if source_code not in ACCEPTED_ONTOLOGIES:
+            print('The Source code entered is invalid.')
+            sys.exit(1)
+
+    #3. Check if source_code and target_code are not the same
+    if source_code == target_code:
+        print("The source and target ontology columns to be mapped are the same.\nSource Ontology: %s\nTarget Ontology: %s" % (source_code,target_code))
+        print("Please specify different ontologies to map on.\n")
+        sys.exit(1)
+
+    return(source_code)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--source-file', required = True, help = 'This is the source file path. The source file must contain one of the ONCOTREE_CODE, ICDO_CODE, NCIT_CODE, UMLS_CODE or HEMEONC_CODE in the file header and it must contain codes corresponding to the Ontology System.',type = str)
+    parser.add_argument('-o', '--target-file', required = True, help = 'This is the path to the target file that will be generated. It will contain the ontology mappings of source code in <source-file> to <target-code>.',type = str)
+    parser.add_argument('-t', '--target-code', required = True, help = "This is the target ontology code that the script will attempt to map the source file ontology code to. It must be one of the ONCOTREE_CODE, ICDO_CODE, NCIT_CODE, UMLS_CODE or HEMEONC_CODE.",type = str)
+    args = parser.parse_args()
+
+    target_file = args.target_file
+    target_code = args.target_code.upper()
+
+    source_file = pd.read_csv(args.source_file,comment='#',sep='\t',header=0,keep_default_na=False).applymap(str) #HEMEONC_CODES are numbers.
+    source_code = extract_source_code_validate_arguments(source_file,target_code)
+
+    data = requests.get(ONTOLOGY_MAPPINGS_FILE_URL).content
+    mappings_file = pd.read_csv(io.StringIO(data.decode('utf-8')),sep='\t',header=0,keep_default_na=False).applymap(str)
+
+    #For case insensitive merge.
+    source_file[source_code] = source_file[source_code].str.upper()
+    mappings_file[source_code] = mappings_file[source_code].str.upper()
+
+    mapped_data = pd.merge(source_file,mappings_file[[source_code,target_code]],on=source_code, suffixes= ('_source_file',''),sort=False,how='left').fillna('')
+    print("Mapped the %s to %s.." % (source_code,target_code))
+
+    add_comments_column_and_log_data(mapped_data,target_file,source_code,target_code,source_file)
+    print("\nThe ontology mappings are written to: %s" % (target_file))
+    print("The mapping summary is written to : %s" % (os.path.splitext(target_file)[0] + "_summary.html"))
```

Comment:
we overwrite the old summary.html file I suppose .. which is fine.

Reply:
I guess it's fine?